inc/ustream.h

```diff
@@ -101,6 +101,34 @@ MOCKABLE_FUNCTION(, AZ_ULIB_RESULT, az_ustream_concat,
         AZ_USTREAM_MULTI_DATA_CB*, multi_data,
         AZ_RELEASE_CALLBACK, multi_data_release);
 
+/**
+  * @brief   Split a ustream at a given position.
+  *
+  *  The split will divide a given ustream into two parts, divided at a passed position. The result of this operation will be two
+  *     ustream instances. The first and original ustream will contain the data from the original position up to but not including 
+  *     the passed <tt>split_pos</tt>. The second ustream will contain the data from the passed <tt>split_pos</tt> up to the end of the
+  *     <tt>ustream_instance</tt> before the function was called. The <tt>split_pos</tt> should be relative to the position returned
```

Comment:
>ustream_instance [](start = 12, length = 16)

Please add some information about the logic position of the first byte of the ustream_instnace_split. Will it start on `0`?

Reply:
added. it will be `split_pos`

---
In reply to: [333761172](https://github.com/Azure/azure-ulib-c/pull/36#discussion_r333761172) [](ancestors = 333761172)